### PR TITLE
add config and unit test. add config file for test.

### PR DESCRIPTION
**Pull request by nkristianto**
add configuration package to read configuration via YAML file
add unit test for configuration
add valid data test and invalid data test to test configuration package

### CODE REVIEW

File: configuration/config.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		return err
+	}
+
+	p, err := byteBuffer.Write(data)
+	if err != nil {
+		return err
+	}
+
+	if p == 0 {
+		return errors.New("failed to read. 0 data is readed")
```
Comment (burubur) [inline configuration/config.go]: Readed?
Suppose to be read

**Comment by nkristianto** [comment]
Bukan nya readed ya? 0 data yg dibaca? Gtu bukan?

**Comment by burubur** [comment]
V1 read
v2 read
v3 read

**Comment by nkristianto** [comment]
noted om, thanks

**Pull request merged**